lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -870,6 +877,45 @@ def ca_rotation_complete_in_cluster(self) -> bool:
 
         return rotation_complete
 
+    def ca_and_certs_rotation_complete_in_cluster(self) -> bool:
+        """Check whether the CA rotation completed in all units."""
+        rotation_complete = True
+
+        # the current unit is not in the relation.units list
+        if (
+            self.charm.peers_data.get(Scope.UNIT, "tls_ca_renewing")
+            or self.charm.peers_data.get(
+                Scope.UNIT,
+                "tls_ca_renewed",
+            )
+            or self.charm.peers_data.get(Scope.UNIT, "tls_configured") is not True
+        ):
+            logger.debug("TLS CA rotation ongoing on this unit.")
+            return False
+
+        for relation_type in [
+            PeerRelationName,
+            PeerClusterRelationName,
+            PeerClusterOrchestratorRelationName,
+        ]:
+            for relation in self.model.relations[relation_type]:
+                logger.debug(f"Checking relation {relation}: units: {relation.units}")
+                for unit in relation.units:
+                    if (
+                        "tls_ca_renewing" in relation.data[unit]
+                        or "tls_ca_renewed" in relation.data[unit]
+                        or relation.data[unit].get("tls_configured") != "True"
```

Comment:
This is basically the same condition as in line 891 and should be adjusted to `is not True` as well.

Reply:
Nevermind, was wrong from my side.